Samples/InpaintHTTP/WebHost.cs

```diff
@@ -0,0 +1,131 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Nancy;
+using Nancy.Hosting.Self;
+using Nancy.ErrorHandling;
+using System.Linq;
+using Zavolokas.GdiExtensions;
+using Zavolokas.ImageProcessing.Inpainting;
+using Zavolokas.Structures;
+using System.Threading.Tasks;
+using Zavolokas.ImageProcessing.PatchMatch;
+using System.Drawing;
+using System.IO;
+using System.Threading;
+//using Zavolokas.Utils.Processes;
+
+//TODO: Cleanup the mess above & unused References
+
+namespace InpaintHTTP
+{
+    class WebHost
+    {
+        public WebHost()
+        {
+            NancyHost host = new NancyHost(new Uri("http://localhost:8069"));
+            host.Start();
+            Console.WriteLine("Initialized and started WebHost");
+        }
+    }
+
+    public class MainMod : NancyModule
+    {
+        public MainMod()
+        {
+            Post("/api/inpaint", async x =>
+            {
+                Console.WriteLine("[" + DateTime.Now.ToString("HH:mm:ss.fffff") + "] Incomming request from " + this.Request.UserHostAddress);
+                if (this.Request.Files.Count() > 1)
```

Comment:
It is better to check that the requirement on the amount of files is met and return the error in the beginning. I mean

```csharp
if (files.Count < 2)
{
    return Error;
}

// do the processing
return Result;
```

rather than

```csharp
if (files.count > 1)
{
   // do the processing
  return Result;
}
return Error;
```

Reply:
This is done to make it more readable?